code/items/ammo/TTTAmmo.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+using Sandbox;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_ammo")]
+    public abstract partial class TTTAmmo : Prop
+    {
+        /// <summary>
+        /// String definition of ammo type, should match TTTWeapon.AmmoType
+        /// </summary>
+        public virtual string Name { get; set; }
+        /// <summary>
+        /// Amount of Ammo within Entity.
+        /// </summary>
+        public virtual int Amount { get; set; }
+
+        /// <summary>
+        /// Maximum amount of ammo player can carry in their inventory.
+        /// </summary>
+        public virtual int Max { get; set; }
+
+        [Net]
+        private int CurrentAmmo { get; set; }
+        private int AmmoEntMax { get; set; }
+
+        /// <summary>
+        /// Fired when a player picks up any amount of ammo from the entity.
+        /// </summary>
+        protected Output OnPickup { get; set; }
+
+        public virtual string ModelPath => "models/ammo/ammo_buckshot.vmdl";
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            SetModel(ModelPath);
+            SetupPhysicsFromModel(PhysicsMotionType.Dynamic);
+            CollisionGroup = CollisionGroup.Weapon;
+            SetInteractsAs(CollisionLayer.Debris);
+
+            AmmoEntMax = Amount;
+            CurrentAmmo = Amount;
+        }
+
+        public override void Touch(Entity other)
+        {
+            base.Touch(other);
+
+            if (other is TTTPlayer player && IsServer)
+            {
+                string ammoType = Name.ToLower();
+                Inventory inventory = (Inventory) player.Inventory;
+
+                if (inventory.GetAmmoTypes().Contains(ammoType))
+                {
+                    int playerAmount = inventory.Ammo.Count(ammoType);
+
+                    if (Max >= (playerAmount + Math.Ceiling(CurrentAmmo * 0.25)))
+                    {
+                        int amountGiven = Math.Min(CurrentAmmo, Max - playerAmount);
+                        inventory.Ammo.Give(ammoType, amountGiven);
+                        CurrentAmmo -= amountGiven;
+                        OnPickup.Fire(other);
+
+                        if (CurrentAmmo <= 0 || Math.Ceiling(AmmoEntMax * 0.25) > CurrentAmmo)
+                        {
+                            Delete();
+                        }
+                    }
+                }
+            }
+
+        }
+
+        public void SetCurrentAmmo(int ammo)
+        {
+            CurrentAmmo = ammo;
+        }
+
+        public override void TakeDamage(DamageInfo info)
+        {
+            //Excerpt from Sandbox.BasePhysics | Removes the ability to remove entity health and destroy, but will still calculate appropriate physics for damage.
+            //Unsure how Prop.Invulnerable works in comparison to this. Might be a better solution.
```

Comment:
Is this comment still needed? https://github.com/TTTReborn/tttreborn/pull/167/files#r682210354